kinesis/consumer.py

```diff
@@ -387,15 +387,16 @@ async def __anext__(self):
             try:
                 item = self.queue.get_nowait()
 
+                if isinstance(item, (bytes, bytearray)):
+                    return item
```

Comment:
I think the item here is post-deserialization, right? So if you want to read/write objects of some custom class, at this point the item is an instance of MyClass, and which is not bytes/bytearray, and so this won't help. Does that sound right?

Basically I'd like to use a custom serializer with our classes so that i can

    class MySerializer(Serializer):
        """defines serialize/deserialize for MyClass objects"""
        ...

    obj = MyClass('foo', 'bar')
    producer.put(obj)
    ...
    async for obj in consumer:
        assert isinstance(obj, MyClass)

And judging by the error message I received, I think the item here has already been deserialized into MyClass ( or TestMessage, in this case ) and is not the bytes produced by the serializer. Though it may be that I was doing something else wrong.

    File "/usr/local/lib/python3.8/site-packages/kinesis/consumer.py", line 390, in __anext__
    if item and "__CHECKPOINT__" in item:
    TypeError: argument of type 'TestMessage' is not iterable

That said, having a BytesSerializer and forcing the serde out into userland and being able to put(bytes) and get bytes back is *also* a nice feature which is not possible due to this same issue. So I think this change helps use ByteSerializers but I was looking to use serializers for instances of our classes.

Reply:
oops yeah sorry, changed it to check its a dict. seems even easier?